src/pages/task/taskTabs/logs/LogTypes.tsx

```diff
@@ -192,8 +194,8 @@ const useRenderBody: React.VFC<{
   const taskAnalytics = useTaskAnalytics();
   const updateQueryParams = useUpdateURLQueryParams();
   const noLogs = !!((error && !data) || !data.length);
-  const onChangeLog = (event: React.ChangeEvent<HTMLInputElement>): void => {
-    const nextLogType = event.target.value as LogTypes;
+  const onChangeLog = (value: string): void => {
```

Comment:
> Because we are on an old component version, it is a little wonky, but not unusable by any means.

I actually am not sure that this is why! I believe the animation is a little messed up by how the `currentLog` state is managed. I noticed that if you console.log `currentLog`, clicking on a new log type prints/rerenders the initial log type before printing the new type. Maybe that's related to how the query param hook works? Regardless, when I set up a simpler state with the component, it transitions smoothly. E.g.

```js
const [currentLog, setCurrentLog] = useState("all");

return (
  <SegmentedControl onChange={setCurrentLog} value={currentLog}>
    {...}
  </SegmentedControl>
);
```

Not sure this is worth the effort to fix but I wanted to flag! Maybe you have some tricks up your sleeve 😉

Reply:
You are so right, the fool doth think he is wise 😔

I ended up moving the LogHeader (i.e. the segmented control options, the buttons) up to the parent component because otherwise they re-render with every log type change, which I don't think makes a lot of sense.